### ISSUE TITLE
[eme] playback-persistent-license-events.js does not allow for a server certificate request

### ISSUE THREAD

**Issue by ddorwin** [w3c/web-platform-tests#4011]
https://w3c-test.org/encrypted-media/drm-mp4-playback-persistent-license-events.html uses `playback-persistent-license-events.js`, which currently expects the following actions and events:
```
[
    'generaterequest',
    'license-request',
    'license-request-response',
    'updated',
    'keystatuseschange',
    'playing',
    'remove',
    'keystatuseschange',
    'license-release',
    'license-release-response',
    'closed-promise',
    'updated'
]
```
Chrome OS 54.0.2840.51 has the following results:

>0: "generaterequest"
**1: "license-request"**
**2: "license-request-response"**
**3: "updated"**
4: "license-request"
5: "license-request-response"
6: "updated"
7: "keystatuseschange"
8: "playing"
9: "license-release"
10: "license-release-response"
11: "updated"

The Chrome OS results show a second pair of license exchange messages. This is likely caused by a request for a server certificate in 1-3, meaning 4-6 are the events the test actually expects.

<br>For reference, here's an analysis of the remaining results. We'll verify the expectations and check a newer version of Chrome OS.
* After that, the events match through "playing".
* We do not see the remove() promise resolved or the corresponding keystatuseschange on Chrome OS.
* Then the license release exchange works as expected except the closed promise is not resolved.

**Comment by ddorwin** [comment]
I suspect this would be an issue with any test that records such actions/events on a client that requires a server certificate. We just haven't analyzed all other tests on such clients.

Ideally, we can find a generic pattern to handle this. or example, if "license-request" is received and the last entry is "updated", remove the last three events. For robustness, we could compare the three events and limit this to happening once per expected license request.

**Comment by mwatson2** [comment]
PR #4013

**Comment by ddorwin** [comment]
Note that while the fix in #4013 should correctly handle successes, it can be confusing in the case of failures where there are less "real" events than expected, as is the case in the Chrome OS example above.

In order to correctly skip the "duplicate" events, Chrome OS would need to produce 15 events to allow 15 - 8 = 7 events to be checked.

**Comment by mwatson2** [comment]
@ddorwin Ok, so how about #4014 ?
